plugins/amazonq/chat/jetbrains-community/src/software/aws/toolkits/jetbrains/services/cwc/inline/InlineChatController.kt

```diff
@@ -438,89 +543,33 @@ class InlineChatController(
-                    val changeEndOffset = getLineEndOffset(document, row)
-                    val rangeMarker = highlightString(editor, changeStartOffset, changeEndOffset, false)
-                    partialUndoActions.add {
-                        editor.markupModel.removeAllHighlighters()
-                    }
-                    partialAcceptActions.add {
-                        if (rangeMarker.isValid) {
-                            deleteString(document, rangeMarker.startOffset, rangeMarker.endOffset)
-                        }
-                        editor.markupModel.removeAllHighlighters()
-                    }
-                }
-
-                DiffRow.Tag.INSERT -> {
-                    val newLineInserted = insertNewLineIfNeeded(row, editor)
-                    val insertOffset = getLineStartOffset(document, row)
-                    val textToInsert = unescape(diffRow.newLine) + "\n"
-                    val rangeMarker = insertString(editor, insertOffset, textToInsert)
-                    partialUndoActions.add {
-                        if (rangeMarker.isValid) {
-                            deleteString(document, rangeMarker.startOffset, (rangeMarker.endOffset + newLineInserted).coerceAtMost(document.textLength))
-                        }
-                        editor.markupModel.removeAllHighlighters()
-                    }
-                    partialAcceptActions.add {
-                        editor.markupModel.removeAllHighlighters()
-                    }
-                }
-
-                else -> {
-                    val changeOffset = getLineStartOffset(document, row)
-                    val changeEndOffset = getLineEndOffset(document, row)
-                    val oldTextRangeMarker = highlightString(editor, changeOffset, changeEndOffset, false)
-                    partialAcceptActions.add {
-                        if (oldTextRangeMarker.isValid) {
-                            deleteString(document, oldTextRangeMarker.startOffset, oldTextRangeMarker.endOffset)
-                        }
-                        editor.markupModel.removeAllHighlighters()
-                    }
-                    val insertOffset = getLineEndOffset(document, row)
-                    val newLineInserted = insertNewLineIfNeeded(row, editor)
-                    val textToInsert = unescape(diffRow.newLine) + "\n"
-                    val newTextRangeMarker = insertString(editor, insertOffset, textToInsert)
-                    partialUndoActions.add {
-                        if (newTextRangeMarker.isValid) {
-                            deleteString(document, newTextRangeMarker.startOffset, newTextRangeMarker.endOffset + newLineInserted)
-                        }
-                        editor.markupModel.removeAllHighlighters()
-                    }
+    private fun removeHighlighter(editor: Editor) {
+        ApplicationManager.getApplication().invokeAndWait {
```

Comment:
does this actually need to block

Reply:
I think so because we will add new highlights after clearing